src/provider/provider_coarse.c

```diff
@@ -549,6 +843,11 @@ static enum umf_result_t coarse_memory_provider_alloc(void *provider,
                coarse_provider->used_size, coarse_provider->alloc_size);
     }
 
+    rv = allocated_blocks_add(coarse_provider->allocated_blocks, new_block);
```

Comment:
please move this closer to umfMemoryProviderAlloc. Do we still need the L817?

Reply:
Done.

> Do we still need the L817?
What do you mean?